### PR TITLE
priority of tasks can be declared via the API 

### LINKED ISSUE #11114: Can't set task priority via the API
## Observed behavior

If you enqueue a task from the backend, you can specify its priority. However, if you enqueue a task via the tasks API you can't. It would be nice to be able to do this so that the frontend can prioritize the tasks it's scheduling.

## Errors and logs

There are no errors. Tasks scheduled via the API use the priority declared in the `register_task` decorator.

## Expected behavior

The priority of each task can be specified in the list sent to the tasks API.

## User-facing consequences

The way most users schedule tasks is via the API as called by the frontend. User scheduled tasks are likely of higher importance than other non-user scheduled tasks, so it would be nice for the frontend to specify a higher priority. Alternatively, the user may know that the task they're scheduling isn't that important to them and can be run at some later time. The frontend would specify a lower priority in that case.

## Steps to reproduce

Start Kolibri with 0 regular priority workers and 1 high priority worker. Go to the device content management page and start several tasks of different types. Then go the device task panel and note that the order that they start is not the same as the order they were scheduled.

## Context

0.16.0-beta1, Linux

### PR DESCRIPTION
**Pull request by im-NL**
## Summary

* fixes #11114 
* priority of tasks can be declared via the API 

## Reviewer guidance

* all tests are passing and priority option has been added to the enqueue_args serializer. 
* I could not completely test the PR manually and would need either someone else to test or get guidance on how to test this change 
----

## Testing checklist

- [ ] Contributor has fully tested the PR manually

## PR process

- [ ] PR has the correct target branch and milestone
- [ ] PR has 'needs review' or 'work-in-progress' label
- [ ] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

File: kolibri/core/tasks/validation.py [hunk 1]
```diff
@@ -15,6 +17,7 @@ class EnqueueArgsSerializer(serializers.Serializer):
     retry_interval = serializers.IntegerField(
         required=False, allow_null=True, min_value=0
     )
+    priority = serializers.IntegerField(required=False, allow_null=False)
```
Comment (rtibbles) [inline kolibri/core/tasks/validation.py]: Could also use the `ChoiceField` here, with `Priority.Priorities` as the `choices` kwarg, which would mean that you wouldn't need the additional validation logic below: https://www.django-rest-framework.org/api-guide/fields/#choicefield

File: kolibri/core/tasks/api.py [hunk 2]
```diff
@@ -150,7 +150,7 @@ def _enqueue_job_based_on_enqueue_args(self, registered_task, job, enqueue_args)
                 enqueue_args["enqueue_at"],
                 job,
                 queue=registered_task.queue,
-                priority=registered_task.priority,
+                priority=enqueue_args.get("priority", registered_task.priority),
```
Comment (rtibbles) [inline kolibri/core/tasks/api.py:153]: Excellent, this looks like the right way to handle this!

**Review by rtibbles — COMMENTED** [review]
This looks like the right change - one possible tweak to tighten up the implementation.

Would also be good to add additional tests to the EnqueueArgsCreateTestCase here: https://github.com/learningequality/kolibri/blob/release-v0.16.x/kolibri/core/tasks/test/test_api.py#L598 to validate this update.

File: kolibri/core/tasks/api.py (hunk 2 above)
Comment (im-NL) [reply kolibri/core/tasks/api.py:153]: Thank you!

**Review by rtibbles — APPROVED** [review]
Excellent! This is safe to merge to release-v0.16.x so I will do a little rebasing here to retarget it.

**Pull request merged**